### PR TITLE
Enable sumKeys option to sum more than one key when data are grouped

### LINKED ISSUE #47: FR: Add option to use different property as rect area
A nice feature to be considered is the possibility to express rect sizes with another property.

> For instance, using the US states demo: area dimension property used for rect surface and the population property used as color.

From https://github.com/kurkle/chartjs-chart-treemap/issues/45#issuecomment-938611142

### PR DESCRIPTION
**Pull request by stockiNail**
Fix #47

New `sumKeys` option could be an array of keys of the objects used in tree and consumed grouping the data.
The result of sums of these additional keys will be stored in data object in `vs` key as object, where the keys are the same that `sumKeys` option has provided.

```javascript
{
  "x": 0,
  "y": 32,
  "w": 714,
  "h": 682,
  "a": 486948,
  "v": 50,
  "vs": {
    "pop": 40
  },
  "s": 50,
  "g": "A",
  "l": 0
}
```

### TODO

- [x] test cases
- [x] types
- [x] documentation

### CODE REVIEW

**Comment by kurkle** [comment]
How about just allowing an array as `key`?
And internally, just make it always be an array, so the calculation is cleaner.
Just a thought, does not have to be that way.

**Comment by stockiNail** [comment]
I thought to use ah array but  in this case we must assume the first element of the array is the key to use for grouping and drawing.
Personally I dont like this convention based on the position of an element in an array. I will have a look anyway next week if is not urgent

**Comment by kurkle** [comment]
The "position 1" convention can be kept internal and should only be used if the code is cleaner that way. Maybe sumKeys would be better name for it? Or additionalSumKeys.

**Comment by stockiNail** [comment]
> Maybe sumKeys would be better name for it? Or additionalSumKeys.

Done, in `sumKeys`.

**Comment by stockiNail** [comment]
> The "position 1" convention can be kept internal and should only be used if the code is cleaner that way.

Let me review better the code in order to merge key and sumKeys in an array (and then arguments for all other functions which are receiving that).

**Comment by stockiNail** [comment]
@kurkle when you have time, can you have a look if sounds better now? PR is still in draft.

**Review by kurkle — COMMENTED** [review]
Yeah, this way looks good to me!

File: src/controller.js [hunk 1]
```diff
@@ -101,21 +100,21 @@ export default class TreemapController extends DatasetController {
     const dataset = this.getDataset();
     const {data} = this.getMeta();
     const groups = dataset.groups || (dataset.groups = []);
-    const key = dataset.key;
+    const keys = [dataset.key || ''].concat(dataset.sumKeys || (dataset.sumKeys = []));
```
[the remark this answers is not in the document]
Comment (stockiNail) [reply src/controller.js:103]: I agree. I'm changing both but I want to check for "groups" if there is any reference to that, assuming is not undefined.

[the remark this answers is not in the document]
Comment (stockiNail) [reply src/controller.js]: checked and updated both

File: docs/usage.md [hunk 2]
```diff
@@ -92,14 +93,15 @@ All these values, if `undefined`, fallback to the scopes described in [option re
 
 | Name | Description
 | ---- | ----
+| `sumKeys` | Define multiple keys to add additional sums, on top of the `key` one, for scriptable options use.
```
Comment (kurkle) [inline docs/usage.md:96]: Is there a reason to have this first? If not, order alphabetically

File: test/fixtures/grouped/treeAdditionalKeys.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+const data = {
```
Comment (kurkle) [inline test/fixtures/grouped/treeAdditionalKeys.js:1]: Should rename these files to match the option name

File: docs/usage.md (hunk 2 above)
Comment (stockiNail) [reply docs/usage.md:96]: No reason... I forgot to move it when I have changed the option name :( Sorry

File: test/fixtures/grouped/treeAdditionalKeys.js (hunk 3 above)
Comment (stockiNail) [reply test/fixtures/grouped/treeAdditionalKeys.js:1]: Yes! For the same above reason :(

**Review by kurkle — APPROVED** [review]

**Pull request merged**